src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -341,6 +351,8 @@ private void BuildLinkTag(TagHelperAttributeList attributes, TagHelperContent bu
                 if (AppendVersion == true &&
                     string.Equals(attribute.Name, HrefAttributeName, StringComparison.OrdinalIgnoreCase))
                 {
+                    EnsureFileVersionProvider();
```

Comment:
I'm hoping this can be removed entirely. But definitely should not be moved inside a loop.